tests/bluemira/builders/EUDEMO/test_tools.py

```diff
@@ -174,3 +184,57 @@ def greater_or_close(values: np.ndarray, limit: float, **kwargs):
         return np.all(
             np.logical_or(values >= limit, np.isclose(values, limit, **kwargs))
         )
+
+
+class TestPatterning:
+    fixture = [
+        (3, 16, 0.0),
+        (3, 16, 0.1),
+        (10, 20, 0.05),
+        (2, 1, 1),
+    ]
+
+    @pytest.mark.parametrize("n_segments, n_sectors, gap", fixture)
+    def test_revolved_silhouette(self, n_segments, n_sectors, gap):
+        p = make_polygon({"x": [4, 5, 5, 4], "y": 0, "z": [-1, -1, 1, 1]}, closed=True)
+        face = BluemiraFace(p)
+
+        shapes = pattern_revolved_silhouette(face, n_segments, n_sectors, gap)
+
+        assert len(shapes) == n_segments
+
+        self._assert_volumes_same(shapes)
+        self._assert_distances_right(shapes, gap)
```

Comment:
Could this assertion be more descriptive/specific? The fact we want the distances to be 'right' is probably a given 😄. 

I'd probably try and avoid making assertions in loops as well - it can be a bit of a pain debugging where in the loop you've exited. I would return all the distances from the private helper, then assert they're all equal to gap.

Maybe something like:
```python
np.testing.assert_almost_equal(self._distances_between(shapes), gap)
```

Similar applies to the looping in `_assert_volumes_same`